### PR TITLE
EOS-22342 : Adding haproxy service handler

### PR DESCRIPTION
**Pull request by shahidSTX**
Adding haproxy service handler and command to start haproxy inside container.

Signed-off-by: Mohammed Shahid <mohammed.shahid@seagate.com>

# Problem Statement
- Add haproxy to single entry-point script

# Design
-  Documented design changes [here](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/639434861/K8+single+node+deployment+without+systemd+on+VM+Only+for+S3+Internal+Purpose#1.-Configuration-changes-in-containers.)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/provisioning/s3_start [hunk 1]
```diff
@@ -58,6 +58,8 @@ def main():
     elif args.service == 's3bgconsumer':
       # Add action and remove pass
       pass
+    elif args.service == 'haproxy':
+      os.system("/usr/sbin/haproxy -Ws -f /etc/cortx/s3/haproxy.cfg -p /run/haproxy.pid &")
```
Comment (sachitanands) [inline scripts/provisioning/s3_start:62]: should we add one isFileExists() validation for these parameters of above command command ?

Comment (shahidSTX) [reply scripts/provisioning/s3_start:62]: @t7ko-seagate Should we add this check suggested by @sachitanands or will provisioner do this for us?

Comment (t7ko-seagate) [reply scripts/provisioning/s3_start:62]: No I don't think we need to check for these files/params now.  Miniprovisioner must ensure they are created.

Comment (t7ko-seagate) [inline scripts/provisioning/s3_start:62]: Please remove the `&` at the end -- single entry point script must "block" and never return, until haproxy itself exits.

**Review by t7ko-seagate — CHANGES_REQUESTED** [review]

File: scripts/provisioning/s3_start (hunk 1 above)
Comment (t7ko-seagate) [inline scripts/provisioning/s3_start:62]: Also -- is this really `/run/haproxy.pid`?  I previously saw pid files stored in `/var/run/<process>.pid`.  Please check.

Comment (t7ko-seagate) [reply scripts/provisioning/s3_start:62]: ok checked myself -- it's /run, closing this comment.

**Review by t7ko-seagate — APPROVED** [review]

**Review by bkirunge7 — APPROVED** [review]

File: scripts/provisioning/s3_start [hunk 2]
```diff
@@ -58,6 +61,10 @@ def main():
     elif args.service == 's3bgconsumer':
       # Add action and remove pass
       pass
+    elif args.service == 'haproxy':
+      # TODO: logging path has to come from conf store -- to be fixed during
+      # end-to-end logging integration.
+      os.system("/usr/sbin/haproxy -Ws -f /etc/cortx/s3/haproxy.cfg -p /run/haproxy.pid 2>&1 1>/var/log/cortx/s3/haproxy.log")
```
Comment (sachitanands) [inline scripts/provisioning/s3_start:67]: do we need to create this directory  first ? 
/var/log/cortx/s3/haproxy

Comment (shahidSTX) [reply scripts/provisioning/s3_start:67]: We are doing it at line 50.

**Review by sachitanands — APPROVED** [review]
looks good

**Comment by t7ko-seagate** [comment]
Triggered the job manually:
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/CentOS-7.9.2009/job/s3server/96/

Job failed due to s3bg consumer.  NOTE - the fix here DOES NOT TOUCH BG SERVICE.  I do not understand why it fails.

image

**Comment by t7ko-seagate** [comment]
Re-triggering the job (in case it was intermittent issue).

http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/CentOS-7.9.2009/job/s3server/97

SUCCESS this time:

image

**Pull request merged**